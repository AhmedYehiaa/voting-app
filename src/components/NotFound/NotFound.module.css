.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.status {
  font-size: 100px;
}

.message h2 {
  text-transform: capitalize;
}

.link {
  padding: 20px;
  margin: 30px auto 0;
  display: block;
  max-width: 250px;
  border-radius: 20px;
  background-color: #00cea5;
  color: #fff;
  text-align: center;
  font-weight: bold;
  text-decoration: none;
  text-transform: uppercase;
}
