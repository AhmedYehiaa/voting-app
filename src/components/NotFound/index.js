import React from "react";
import { Link } from "react-router-dom";

import styles from "./NotFound.module.css";

const NotFound = () => {
  return (
    <div className={styles.container}>
      <div className={styles.status}>
        <h1>404</h1>
      </div>
      <div className={styles.errorMessage} data-testid="errorMessage">
        <h2>oops! this page could not be found</h2>
        <Link className={styles.link} to="/" data-testid="link">
          Return to Home Page
        </Link>
      </div>
    </div>
  );
};

export default NotFound;
