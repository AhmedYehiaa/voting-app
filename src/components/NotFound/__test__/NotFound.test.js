import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import NotFound from "../index";

afterEach(cleanup);

describe("NotFound Component", () => {
  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Router>
        <NotFound />
      </Router>,
      div
    );
  });

  it("renders NotFound page with error message ", () => {
    const { getByTestId } = render(
      <Router>
        <NotFound />
      </Router>
    );
    expect(getByTestId("errorMessage")).toHaveTextContent(
      "oops! this page could not be found"
    );
  });

  it("renders Not found page with Link to home page ", () => {
    const { getByTestId } = render(
      <Router>
        <NotFound />
      </Router>
    );
    expect(getByTestId("link")).toBeEnabled();
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <NotFound />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
