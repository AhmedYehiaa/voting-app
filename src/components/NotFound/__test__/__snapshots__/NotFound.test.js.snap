// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="status"
  >
    <h1>
      404
    </h1>
  </div>
  <div
    className="errorMessage"
    data-testid="errorMessage"
  >
    <h2>
      oops! this page could not be found
    </h2>
    <a
      className="link"
      data-testid="link"
      href="/"
      onClick={[Function]}
    >
      Return to Home Page
    </a>
  </div>
</div>
`;
